src/frontend/Ast_to_Mir.ml

```diff
@@ -10,6 +10,21 @@ let trans_fn_kind = function
   | Ast.StanLib -> Middle.StanLib
   | UserDefined -> UserDefined
 
+let get_prob_fun_name name =
+  let new_name =
+    ["_log"; "_lpdf"; "_lpmf"; ""]
+    |> List.map ~f:(( ^ ) name)
+    |> List.filter ~f:Stan_math_signatures.is_stan_math_function_name
+    |> List.hd
+  in
+  match new_name with
+  | Some n -> n
+  | None -> raise_s [%message "No matching functions for" (name : string)]
+
+let%expect_test "distribution name mangling" =
+  print_endline (get_prob_fun_name "normal") ;
+  [%expect {| normal_log |}]
+
 let rec op_to_funapp op args =
   let argtypes =
     List.map ~f:(fun x -> (x.Ast.emeta.Ast.ad_level, x.emeta.type_)) args
```

Comment:
Shouldn't you be special casing on And and Or here to create EAnd and EOr?

Reply:
Yep, that seems right. Good catch, thanks.